system/lib/libc/musl/arch/emscripten/bits/alltypes.h

```diff
@@ -91,9 +91,9 @@ typedef long suseconds_t;
 #if defined(__NEED_pthread_attr_t) && !defined(__DEFINED_pthread_attr_t)
 typedef struct {
     union {
-        int __i[9];
-        volatile int __vi[9];
-        unsigned __s[9];
```

Comment:
I would expect this to match the code in upstream musl:

https://github.com/emscripten-core/musl/blob/85e0e3519655220688e757b9d5bfd314923548bd/include/alltypes.h.in#L85

i.e. it would expect it to with be 14 or 9 for each other..  what am I missing here?

Reply:
Not sure.. we have:
```
#define __SU (sizeof(size_t)/sizeof(int))
#define _a_prio __u.__i[3*__SU+3]
```
so accessing that field causes:
```
system\lib\libc\musl\src\thread\pthread_attr_get.c:22:26: error: array index 9 is past the end of the array (which contains 9 elements)
      [-Werror,-Warray-bounds]
        param->sched_priority = a->_a_prio;
```

Reply:
Can you copy what musl does and do `sizeof(long)==8?14:9` .. for consistency with upstream?

Reply:
that code was never in there and would give us suddenly 14 fields in wasm32, which also would require our JS bindings to further become conditional on wasm32/64, see e.g. https://github.com/emscripten-core/emscripten/pull/15229/files#diff-31ce84d9e9748b123befe225a55c67718b58011f9b0a099c821e5c5f0019b1fa

Reply:
OK fair enough.  This change lgtm for now then.

Reply:
Wait, but don't the JS bindings already need to have different offsets for wasm64 since `_a_prio` is different for wasm64 (for example)

Reply:
not sure if we access `_a_prio` in JS (we access it in C), but we access the canvas names field which is currently at a single offset in both. See link above.

Reply:
Hmm, it does look like we access _a_stacksize, _a_stackaddr and _a_detach using hardcoded constants in the JS code:

https://github.com/emscripten-core/emscripten/blob/ec42ea2809c4a0e2b76e831d91f8241e57d63197/src/library_pthread.js#L778-L780

These probably need to be fixes (although not necessarily as part of this PR).

The reference to the canvas is not longer hardcoded and uses the struct info so it should be fine, even if the struct layout varies between 32 and 64 (which it does):

https://github.com/emscripten-core/emscripten/blob/ec42ea2809c4a0e2b76e831d91f8241e57d63197/src/library_pthread.js#L667